src/main/java/uk/gov/hmcts/reform/prl/utils/CaseUtils.java

```diff
@@ -969,6 +971,38 @@ public static String getAdditionalApplicationId(WaMapper waMapper) {
         return null;
     }
 
+    public static String getHearingId(WaMapper waMapper) {
+        if (null != waMapper && null != waMapper.getClientContext().getUserTask()) {
+            if (null != waMapper.getClientContext().getUserTask().getTaskData().getAdditionalProperties()) {
+                return waMapper.getClientContext().getUserTask().getTaskData().getAdditionalProperties().getHearingId();
+            }
+        }
+        return null;
+    }
+
+    public static Optional<Long> getHearingId(WaMapper waMapper, Map<String, Object> caseDataUpdated) {
+        Optional<Long> taskHearingId = ofNullable(waMapper)
+            .filter(value ->
+                        new ObjectMapper().convertValue(
+                            caseDataUpdated.get(IS_INVOKED_FROM_TASK),
```

Comment:
This might throw null pointer as I couldn't find it setting to No.

Reply:
I use your method, good idea to interchange equal condition